carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -36,1360 +36,1488 @@
-      }
+        // all maneuver types in same signal group is currently expected to share signal timing, so only 0th index is used when setting states
+        if (current_movement_state.movement_event_list.empty())
+        {
+          ROS_DEBUG_STREAM("Movement_event_list is empty . intersection_id: " << (int)curr_intersection.id.id << ", and signal_group_id: " << (int)current_movement_state.signal_group);
+          continue;
+        }
 
-      //if same data as last time:
-      //where state is same and timestamp is equal or less, skip
-      if (traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].size() > 0 && traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().second ==
-          static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state) &&
-          traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().first >= min_end_time) 
-      {
-        continue;
-      }
+        // raw min_end_time in seconds measured from the most recent full hour
+        boost::posix_time::ptime min_end_time = lanelet::time::timeFromSec(current_movement_state.movement_event_list[0].timing.min_end_time);
+        auto received_state = static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state);
 
-      // if received same state as last time, but with updated time, just update the time of last state
-      // skip setting state until received a new state that is different from last recorded one
-      if (traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].size() > 0 && traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().second ==
-          static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state) &&
-          traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().first < min_end_time)
-      {
-        ROS_DEBUG_STREAM("Updated time for state: " << static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state)  << ", with time: " 
-                          << std::to_string(lanelet::time::toSec(min_end_time)));
-        traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().first = min_end_time;
-        continue;
-      }
+        if (curr_intersection.moy_exists) //account for minute of the year
+        {
+          auto inception_boost(boost::posix_time::time_from_string("1970-01-01 00:00:00.000")); // inception of epoch
+          auto duration_since_inception(lanelet::time::durationFromSec(ros::Time::now().toSec()));
+          auto curr_time_boost = inception_boost + duration_since_inception;
```

Comment:
Why is it necessary to add the time to the current epoch? If the time is already UTC shouldn't this always give an equal value?

Reply:
its goal is merely a conversion from `posix_duration` to `posix_time`. 
Inception_boost is `posix_time`
ros::Time::now().toSec() is converted to duration easiest. So we add time + duration to get time again.

Reply:
Kinda strange way to go about it, but in the interest of time I think we can leave it as is.